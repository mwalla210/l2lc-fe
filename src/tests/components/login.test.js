import React from 'react'
import renderer from 'react-test-renderer'
import Login from '../../components/login'
jest.mock('../../components/buttonPrimary')

const defaultOptions = {
  event: {
    target: {
      value: 'nameOrPassword'
    },
    preventDefault: jest.fn()
  },
  page: {
    createNewProjMenuItem: jest.fn()
  },
  website: {
    login: jest.fn(),
    updateUsername: jest.fn(),
    updatePassword: jest.fn(),
    loginerror: false,
    username:'testuser',
    password:'testpassword',
    loginButtonDisabled: false,
    },
  }

  const alternateOptions = {
    event: {
      target: {
        value: 'nameOrPassword'
      },
      preventDefault: jest.fn()
    },
    page: {
      createNewProjMenuItem: jest.fn()
    },
    website: {
      login: jest.fn(),
      updateUsername: jest.fn(),
      updatePassword: jest.fn(),
      loginerror: true,
      username:'testuser',
      password:'testpassword',
      loginButtonDisabled: false,
      },
    }

describe('Login', () => {
  it ('Renders with snapshot', () => {
    let options = Object.assign({}, defaultOptions)
    const component = renderer.create(
      <Login {...options}/>,
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it ('Renders with snapshot, alternate arguements', () => {
    let options = Object.assign({}, alternateOptions)
    const component = renderer.create(
      <Login {...options}/>,
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('Renders and calls formSubmit', () => {
    let options = Object.assign({}, defaultOptions)
    const component = renderer.create(
      <Login {...options}/>,
    )
    const inst = component.getInstance()

    expect(inst.props.website.login.mock.calls.length).toBe(0)
    inst.formSubmit(inst.props.event)
    expect(inst.props.website.login.mock.calls.length).toBe(1)
  })

  it('Renders and calls onChangeUsername', () => {
    let options = Object.assign({}, defaultOptions)
    const component = renderer.create(
      <Login {...options}/>,
    )
    const inst = component.getInstance()

    expect(inst.props.website.updateUsername.mock.calls.length).toBe(0)
    inst.onChangeUsername(inst.props.event)
    expect(inst.props.website.updateUsername.mock.calls.length).toBe(1)
  })

  it('Renders and calls onChangePassword', () => {
    let options = Object.assign({}, defaultOptions)
    const component = renderer.create(
      <Login {...options}/>,
    )
    const inst = component.getInstance()

    expect(inst.props.website.updatePassword.mock.calls.length).toBe(0)
    inst.onChangePassword(inst.props.event)
    expect(inst.props.website.updatePassword.mock.calls.length).toBe(1)
  })
})
