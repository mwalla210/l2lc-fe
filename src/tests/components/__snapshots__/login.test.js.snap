// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Renders with snapshot 1`] = `
<div>
  <p
    className="container"
    style={
      Object {
        "display": "flex",
        "justifyContent": "center",
      }
    }
  >
    <img
      src="../../img/L2L-Logo-raisedLARGE.png"
      style={
        Object {
          "margin": "30px",
        }
      }
    />
  </p>
  <h1
    style={
      Object {
        "marginBottom": "30px",
      }
    }
  >
    Line2Line Cloud
  </h1>
  <div
    className="row justify-content-center"
  >
    <div
      className="col-lg-12"
    >
      <div
        className="row justify-content-center"
      >
        <div
          className="col-6 alert alert-warning"
          role="alert"
          style={
            Object {
              "display": "none",
            }
          }
        >
          <strong>
            Warning!
          </strong>
           Couldn't authenticate this username and password combination.
        </div>
      </div>
      <form>
        <div
          className="form-group"
        >
          <div
            className="row justify-content-center"
          >
            <div
              className="col-sm-6"
            >
              <input
                className="form-control"
                id="username"
                name="username"
                onBlur={undefined}
                onChange={[Function]}
                placeholder="Username"
                tabIndex="1"
                type="text"
                value="testuser"
              />
            </div>
          </div>
        </div>
        <div
          className="form-group"
        >
          <div
            className="row justify-content-center"
          >
            <div
              className="col-sm-6"
            >
              <input
                className="form-control"
                id="password"
                name="password"
                onBlur={undefined}
                onChange={[Function]}
                placeholder="Password"
                tabIndex="2"
                type="password"
                value="testpassword"
              />
            </div>
          </div>
        </div>
        <div
          className="form-group"
        >
          <div
            className="row justify-content-center"
          >
            <div
              className="col-sm-6"
            />
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;

exports[`Login Renders with snapshot, alternate arguements 1`] = `
<div>
  <p
    className="container"
    style={
      Object {
        "display": "flex",
        "justifyContent": "center",
      }
    }
  >
    <img
      src="../../img/L2L-Logo-raisedLARGE.png"
      style={
        Object {
          "margin": "30px",
        }
      }
    />
  </p>
  <h1
    style={
      Object {
        "marginBottom": "30px",
      }
    }
  >
    Line2Line Cloud
  </h1>
  <div
    className="row justify-content-center"
  >
    <div
      className="col-lg-12"
    >
      <div
        className="row justify-content-center"
      >
        <div
          className="col-6 alert alert-warning"
          role="alert"
        >
          <strong>
            Warning!
          </strong>
           Couldn't authenticate this username and password combination.
        </div>
      </div>
      <form>
        <div
          className="form-group"
        >
          <div
            className="row justify-content-center"
          >
            <div
              className="col-sm-6"
            >
              <input
                className="form-control"
                id="username"
                name="username"
                onBlur={undefined}
                onChange={[Function]}
                placeholder="Username"
                tabIndex="1"
                type="text"
                value="testuser"
              />
            </div>
          </div>
        </div>
        <div
          className="form-group"
        >
          <div
            className="row justify-content-center"
          >
            <div
              className="col-sm-6"
            >
              <input
                className="form-control"
                id="password"
                name="password"
                onBlur={undefined}
                onChange={[Function]}
                placeholder="Password"
                tabIndex="2"
                type="password"
                value="testpassword"
              />
            </div>
          </div>
        </div>
        <div
          className="form-group"
        >
          <div
            className="row justify-content-center"
          >
            <div
              className="col-sm-6"
            />
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;
